qualtran/_infra/gate_with_registers.py

```diff
@@ -157,12 +157,15 @@ def decompose_bloq(self) -> 'CompositeBloq':
             A CompositeBloq containing the decomposition of this Bloq.
 
         Raises:
-            NotImplementedError: If there is no decomposition defined; namely: if
+            DecomposeNotImplementedError: If there is no decomposition defined; namely: if
                 `build_composite_bloq` returns `NotImplemented`.
         """
         from qualtran.cirq_interop._cirq_to_bloq import decompose_from_cirq_op
 
-        return decompose_from_cirq_op(self, decompose_once=True)
+        try:
+            return Bloq.decompose_bloq(self)
+        except DecomposeNotImplementedError:
+            return decompose_from_cirq_op(self, decompose_once=True)
```

Comment:
can you please update this methods docstring to describe the fallback behavior there (and why I would want it)

Reply:
added a detailed docstring